src/Examine/LuceneEngine/Search/LuceneSearchQueryBase.cs

```diff
@@ -500,5 +515,12 @@ public override string ToString()
         {
             return $"{{ Category: {Category}, LuceneQuery: {Query} }}";
         }
+
+        public abstract IBooleanOperation SelectFields(params string[] fieldNames);
```

Comment:
This becomes a breaking change too 😭  The way around that is to make these virtual that throw NotImplementedException